packages/suite/src/components/suite/SecurityCheck/useReportDeviceCompromised.ts

```diff
@@ -26,34 +32,37 @@ const useCommonData = () => {
 };
 
 const useReportRevisionCheck = () => {
-    const { device } = useDevice();
     const commonData = useCommonData();
-
-    const revisionCheck = isDeviceAcquired(device)
-        ? device.authenticityChecks?.firmwareRevision
-        : null;
-    const isRevisionCheckError = revisionCheck && !revisionCheck.success;
-    const revisionCheckError = isRevisionCheckError ? revisionCheck.error : null;
+    const revisionCheckError = useSelector(selectFirmwareRevisionCheckError);
 
     useEffect(() => {
-        if (!isRevisionCheckError) return;
-        reportCheckFail('revision', { ...commonData, revisionCheckError });
-    }, [commonData, isRevisionCheckError, revisionCheckError]);
+        if (revisionCheckError !== null) {
+            reportCheckFail('revision', { ...commonData, revisionCheckError });
+        }
+    }, [commonData, revisionCheckError]);
 };
 
 const useReportHashCheck = () => {
     const { device } = useDevice();
     const commonData = useCommonData();
 
+    // `errorPayload` must also be extracted, which is why `selectFirmwareHashCheckError` would be impractical
     const hashCheck = isDeviceAcquired(device) ? device.authenticityChecks?.firmwareHash : null;
     const isHashCheckError = hashCheck && !hashCheck.success;
     const hashCheckError = isHashCheckError ? hashCheck.error : null;
     const hashCheckErrorPayload = isHashCheckError ? hashCheck.errorPayload : null;
 
     useEffect(() => {
-        if (!isHashCheckError) return;
+        if (!hashCheckError) return;
+        if (
+            isArrayMember(hashCheckError, skippedHashCheckErrors) &&
+            !isArrayMember(hashCheckError, skippedButReportedHashCheckErrors)
+        ) {
+            return;
+        }
+
```

Comment:
`useReportRevisionCheck` is now much nicer thanks to the new selector that gets error regardless of `enabled`.
But `useReportHashCheck` does not use it, as it also needs payload, so I have to do the whole decision process again. 
:thought_balloon: I could create `selectHashCheckErrorPayload` but sounds like overkill, what do you think?